winit-core/src/event_loop.rs

```diff
@@ -0,0 +1,285 @@
+use std::fmt::{self, Debug};
```

Comment:
Nit: A bit unfortunate that this large diff will have to go in the same commit as everything else here.

I'd prefer a separate PR which moves `ActiveEventLoop`, `ControlFlow` etc. to e.g. `winit/src/event_loop/core.rs`, so that the diff in this PR can be smaller (I can do it if you want me to?)

Similarly for `ApplicationHandlerExtMacOS`, though it's less of an issue there.

Reply:
I don't think it matters though, I generally wanted to just merge the way commits are organized right now, though if you insist on squashing, we can just leave those with the actual diff as a dedicated commits.

The thing is that the order in which commits are is important, like you can not really change stuff to be earlier due to imports and such.

Reply:
Fine with it given that the PR isn't squashed.